### PR TITLE
feat(costmap_generator): activate by location

### PR DESCRIPTION
**Pull request by HiroIshida**
## Description

This PR enables activation of costmap generator by ego vehicle location. activation by location or activation by scenario can be selected by rosparam. The motivation is as follows.

Originally, costamp generator is  activated when the scenario is parking. However, there exists cases that does not fit in to the former activation criterion. For example in automated parking task, a vehicle must cruise around the lanes inside a parking lot in a `lane driving` phase, then find a vacant parking space. In such case, costmap generator must be activated even when `lane driving` scenario. 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/costmap_generator/README.md
Comment (yukkysaito) [inline planning/costmap_generator/README.md:37]: please format :+1:

Comment (HiroIshida) [reply planning/costmap_generator/README.md:37]: Fixed!

**Review by yukkysaito — DISMISSED** [review]

**Comment by HiroIshida** [comment]
@yukkysaito 
Thanks for your review
I want to make another PR that depends on this PR, so it would be great if you could merge this unless I have to do some other actions before merging

**Comment by yukkysaito** [comment]
@HiroIshida Can you check DCO in CI

**Comment by HiroIshida** [comment]
@yukkysaito 
I'm sorry, I added the signed-off in the commits.

**Review by yukkysaito — APPROVED** [review]

**Pull request merged**
